src/main/java/seedu/address/model/person/Budget.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Buyer's budget in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBudget(String)}
+ */
+public class Budget {
+
+    public static final String MESSAGE_CONSTRAINTS = "Budget should be a positive number.";
+    public static final String VALIDATION_REGEX = "\\d+(\\.\\d+)?";
+    public final String value;
+
+    /**
+     * Constructs a {@code Budget}.
+     *
+     * @param budget A valid budget amount.
+     */
+    public Budget(String budget) {
+        requireNonNull(budget);
+        checkArgument(isValidBudget(budget), MESSAGE_CONSTRAINTS);
+        value = budget;
+    }
+
+    /**
+     * Returns true if a given string is a valid budget amount.
+     */
+    public static boolean isValidBudget(String test) {
+        return test.matches(VALIDATION_REGEX) && Double.parseDouble(test) >= 0;
```

Comment:
Can a budget be $0? And is there a minimum and maximum a budget can be?

Things to consider, depends on our team requirement (need to ask them).